js/google_ima.js

```diff
@@ -1212,35 +1256,55 @@ require("../html5-common/js/utils/utils.js");
           case eventType.THIRD_QUARTILE:
             _onAdMetrics(adEvent);
             break;
+          case eventType.VOLUME_CHANGED:
+          case eventType.VOLUME_MUTED:
+            if (this.videoControllerWrapper)
+            {
+              this.videoControllerWrapper.raiseVolumeEvent();
+            }
+            break;
+          case eventType.DURATION_CHANGE:
+            if (this.videoControllerWrapper)
+            {
+              this.videoControllerWrapper.raiseDurationChange(this.getCurrentTime(), this.getDuration());
+            }
+            break;
           default:
             break;
         }
+      });
 
-        if (this.videoControllerWrapper)
+      /**
+       * Starts a timer that will provide an update to the video controller wrapper of the ad's current time.
+       * We use a timer because IMA does not provide us with a time update event.
+       * @private
+       * @method GoogleIMA#_startTimeUpdater
+       */
+      var _startTimeUpdater = privateMember(function()
+      {
+        _stopTimeUpdater();
+        _timeUpdater = setInterval(_.bind(function()
```

Comment:
Are we able to access the video object of the ad and add a playhead time changed or is that guarded by the SDK?